yt/yt/server/scheduler/operations_cleaner.h

```diff
@@ -11,9 +11,18 @@ namespace NYT::NScheduler {
 
 ////////////////////////////////////////////////////////////////////////////////
 
-struct TArchiveOperationRequest
+struct TRemoveOperationRequest
 {
     TOperationId Id;
+    //! It is guaranteed that all dependent nodes will be successfully
+    //! removed before the operation node is removed itself.
+    std::vector<NCypressClient::TNodeId> DependentNodeIds;
+};
+
+struct TArchiveOperationRequest
+    : public TRemoveOperationRequest
```

Comment:
Archiving and removing an operation are two different steps of the cleaner, but here we mix them here and I'm not sure this is a good idea

I would suggest to name the base struct something like `TCLeanOperationRequestBase` or just `TCLeanOperationRequest`, and `TRemoveOperationRequest` and `TArchiveOperationRequest` will inherit it

Reply:
If I do as you say, I will still have to switch to `TCleanOperationRequest` very soon (like RemoveBatcher will have to store `TCleanOperationRequest`), because you can submit an operation for archivation just by operation id => there is a fetch and we build `TArchiveOperationRequest` from attributes => we have to remove this operation afterwards => we need to get `TRemove/CleanOperationRequest` from `TArchiveOperaitonRequest` and pass it to the methods that actually do the removal.

So I can't really see the point? In our paradigm archivation implies removal afterwards.